### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_person/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -28,22 +29,23 @@ const alex = {
 
 const App = () => (
   <div className="App">
-    <section className="Person">
-      <h2 className="Person__name">My name is Misha</h2>
-      <p className="Person__age">I am 37</p>
-      <p className="Person__partner">My wife&apos;s name is Natasha</p>
-    </section>
+    <Person
+      name={misha.name}
```
Comment (alexandra-protyanova) [inline src/App.js:33]: you can make it like:
`<Person person={misha}/>`

**Review by alexandra-protyanova — CHANGES_REQUESTED** [review]
Add demo link, please

File: src/App.js (hunk 1 above)
Comment (Anna-Tkachenko) [reply src/App.js:33]: Agree, will be better to pass full person object to props
Fix it

File: src/Person.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+const Person = ({
+  name,
+  age,
+  partner,
+}) => (
+  <>
+    <h2>{`${name}`}</h2>
+    <p>{`${age}`}</p>
+    <p>{`${partner}`}</p>
```
Comment (Anna-Tkachenko) [inline src/Person.jsx:12]: You should render info about person in the next format
<section className="Person">
  <h2 className="Person__name">My name is Alex</h2>
  <p className="Person__age">I am 25</p>
  <p className="Person__partner">I am not married</p>
</section>

File: src/Person.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  </>
+);
+
+Person.propTypes = {
```
Comment (Anna-Tkachenko) [inline src/Person.jsx:16]: Add prop types for all fields and default props for non required fields

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
Fix comments

File: src/App.js [hunk 4]
```diff
@@ -28,22 +29,11 @@ const alex = {
 
 const App = () => (
   <div className="App">
-    <section className="Person">
-      <h2 className="Person__name">My name is Misha</h2>
-      <p className="Person__age">I am 37</p>
-      <p className="Person__partner">My wife&apos;s name is Natasha</p>
-    </section>
+    <Person {...misha} />
```
Comment (Anna-Tkachenko) [inline src/App.js:32]: Will be better to define some field in props of Person component and pass here person object
```suggestion
    <Person person={misha} />
```

File: src/Person.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Person = ({
+  name,
+  age,
+  sex,
+  isMarried,
+  partnerName,
+}) => {
```
Comment (Anna-Tkachenko) [inline src/Person.jsx:10]: Here you should accept props object, get from here person field and get needed fields from person using desctructuring

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/Person.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        <p className="Person__age">
+          {`I am ${age}`}
+        </p>
+      )}
+
+      <p className="Person__partner">
+        {
+          isMarried ? `My ${partner}'s name is ${partnerName}` : 'I am not married'
+        }
+      </p>
```
Comment (Anna-Tkachenko) [inline src/Person.jsx:26]:
```suggestion
      <p className="Person__partner">
        {
          isMarried
            ? `My ${partner}'s name is ${partnerName}`
            : 'I am not married'
        }
      </p>
```

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**